apps/wallet-mobile/src/legacy/Dashboard/WithdrawStakingRewards/WithdrawStakingRewards.tsx

```diff
@@ -6,42 +6,51 @@ import {ScrollView, StyleSheet, Text, View} from 'react-native'
 import {Boundary} from '../../../components/Boundary/Boundary'
 import {Button} from '../../../components/Button/Button'
 import {Checkbox} from '../../../components/Checkbox/Checkbox'
+import {useModal} from '../../../components/Modal/ModalContext'
 import {PleaseWaitView} from '../../../components/PleaseWaitModal'
 import {Space} from '../../../components/Space/Space'
 import {Warning} from '../../../components/Warning/Warning'
 import {useSelectedWallet} from '../../../features/WalletManager/common/hooks/useSelectedWallet'
 import globalMessages, {confirmationMessages, ledgerMessages} from '../../../kernel/i18n/global-messages'
+import {useWalletNavigation} from '../../../kernel/navigation'
 import {YoroiWallet} from '../../../yoroi-wallets/cardano/types'
 import {useWithdrawalTx} from '../../../yoroi-wallets/hooks'
 import {YoroiUnsignedTx} from '../../../yoroi-wallets/types/yoroi'
+import {delay} from '../../../yoroi-wallets/utils/timeUtils'
 import {Quantities} from '../../../yoroi-wallets/utils/utils'
 import {useStakingInfo} from '../StakePoolInfos'
 import {ConfirmTx} from './ConfirmTx/ConfirmTx'
 type Props = {
   wallet: YoroiWallet
-  onCancel: () => void
-  onSuccess: () => void
 }
 
-export const WithdrawStakingRewards = ({wallet, onSuccess, onCancel}: Props) => {
+export const WithdrawStakingRewards = ({wallet}: Props) => {
   const strings = useWithdrawStakingRewardsStrings()
-  const [state, setState] = React.useState<
-    {step: 'form'; withdrawalTx: undefined} | {step: 'confirm'; withdrawalTx: YoroiUnsignedTx}
-  >({step: 'form', withdrawalTx: undefined})
+  const {closeModal, openModal} = useModal()
+  const {resetToTxHistory} = useWalletNavigation()
+
+  const handleOnConfirm = async (withdrawalTx: YoroiUnsignedTx) => {
+    closeModal()
+
+    await delay(1000)
```

Comment:
I don't love the artificial delay, is it for aesthetics or does it break without it?
Would it make sense to add a new `switchModal` to useModal, that acts like close+open without it breaking?

Reply:
Definitely the Modal context needs a refactor to deal with an array of modals, instead of a single one.

Reply:
that would be rally good